lib/sig_item_map.ml

```diff
@@ -0,0 +1,77 @@
+open Types
+
+type t = {
+  values_map : value_description String_map.t;
+  modules_map : module_declaration String_map.t;
+  types_map : (type_declaration * Ident.t) String_map.t;
+}
+
+type _ item_type =
+  | Value : value_description item_type
+  | Module : module_declaration item_type
+  | Type : (type_declaration * Ident.t) item_type
```

Comment:
There is a very little bit of type duplication here, that could be solved with a type alias, but we will do that if the types get more complex!